### PR TITLE
[Elements] Database transactions for delete() 

### LINKED ISSUE #4053: [Elements] Database transactions for delete()

### PR DESCRIPTION
**Pull request by brusch**
Resolves #4053

### CODE REVIEW

File: lib/Model/Dao/AbstractDao.php [hunk 1]
```diff
@@ -33,19 +33,27 @@ public function configure()
         $this->db = Db::get();
     }
 
-    public function beginTransaction()
+    public function beginTransaction(bool $isNested = false)
     {
-        $this->db->beginTransaction();
+        // MySQL/MariaDB do not support nested transaction, so we just ignore the transaction if
+        // the transaction is started within another transaction (e.g. when deleting elements)
```
[the remark this answers is not in the document]
Comment (brusch) [reply lib/Model/Dao/AbstractDao.php]: Ah, yes, thanks for the hint, wasnt aware of that 👍

[the remark this answers is not in the document]
Comment (dpfaffenbauer) [reply lib/Model/Dao/AbstractDao.php]: will you remove some of the stuff you did? cause actually, you don't need that. Some of it makes things more complicated than needed

[the remark this answers is not in the document]
Comment (brusch) [reply lib/Model/Dao/AbstractDao.php]: I'm on it 😊

[the remark this answers is not in the document]
Comment (brusch) [reply lib/Model/Dao/AbstractDao.php]: Done by 9279f30874d5e0c6cf3114c2c42bd72c9f2245d7

[the remark this answers is not in the document]
Comment (dpfaffenbauer) [reply lib/Model/Dao/AbstractDao.php]: perfect :)

**Pull request merged**